src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_multi.h

```diff
@@ -0,0 +1,229 @@
+ */
+#ifndef SIMPLEX_TREE_MULTI_H_
+#define SIMPLEX_TREE_MULTI_H_
+
+#include <algorithm>
+#include <gudhi/Simplex_tree.h>
+#include <gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h>
+#include <gudhi/Simplex_tree/multi_filtrations/Line.h>
+
+
+
+
+
+namespace Gudhi::multiparameter {
+/** Model of SimplexTreeOptions.
+ *
+ * Maximum number of simplices to compute persistence is <CODE>std::numeric_limits<std::uint32_t>::max()</CODE>
+ * (about 4 billions of simplices). */
+
+
+struct Simplex_tree_options_multidimensional_filtration {
+public:
+	typedef linear_indexing_tag Indexing_tag;
+	typedef int Vertex_handle;
+	typedef float value_type;
+	using Filtration_value = multi_filtrations::Finitely_critical_multi_filtration<value_type>;
+	typedef std::uint32_t Simplex_key;
+	static const bool store_key = true;
+	static const bool store_filtration = true;
+	static const bool contiguous_vertices = false;
+	static const bool link_nodes_by_label = true;
+	static const bool stable_simplex_handles = false;
+	static const bool is_multi_parameter = true;
+};
+
+
+
+using options_multi = Simplex_tree_options_multidimensional_filtration;
+using options_std = Simplex_tree_options_full_featured;
+using simplextree_std = Simplex_tree<options_std>;
+using simplextree_multi = Simplex_tree<options_multi>;
+
+using multi_filtration_type = std::vector<options_multi::value_type>;
+using multi_filtration_grid = std::vector<multi_filtration_type>;
+
+// Turns a 1-parameter simplextree into a multiparameter simplextree, and keeps the 1-filtration in the 1st axis 
+template<class simplextree_std, class simplextree_multi>
+void multify(simplextree_std &st, simplextree_multi &st_multi, const int num_parameters, const typename simplextree_multi::Options::Filtration_value& default_values={}){
+	typename simplextree_multi::Options::Filtration_value f(num_parameters);
+	for (auto i = 0u; i<std::min(static_cast<unsigned int>(default_values.size()), static_cast<unsigned int>(num_parameters-1));i++)
+		f[i+1] = default_values[i];
+	std::vector<int> simplex;
+	simplex.reserve(st.dimension()+1);
+	for (auto &simplex_handle : st.complex_simplex_range()){
+		simplex.clear();
+		for (auto vertex : st.simplex_vertex_range(simplex_handle))
+			simplex.push_back(vertex);
+
+		if (num_parameters > 0)
+		f[0] = st.filtration(simplex_handle);
+		auto filtration_copy = f;
+		st_multi.insert_simplex(simplex,filtration_copy);
+		
+	}
+}
+
+
+
+// Turns a multi-parameter simplextree into a 1-parameter simplextree
+template<class simplextree_std, class simplextree_multi>
+void flatten(simplextree_std &st, simplextree_multi &st_multi, const int dimension = 0){
+	for (const auto &simplex_handle : st_multi.complex_simplex_range()){
+		std::vector<int> simplex;
+		for (auto vertex : st_multi.simplex_vertex_range(simplex_handle))
+			simplex.push_back(vertex);
+		typename simplextree_multi::Options::value_type f = dimension >= 0 ? st_multi.filtration(simplex_handle)[dimension] : 0;
```

Comment:
What is the use of `dimension` to possibly be negative?

Reply:
put every filtration values of `st` to 0. May be useful to copy a simplextree when we don't care about filtration values.

Reply:
Ok, I see. You should say this in the description of `dimension`, now that you added the doc.